### PR TITLE
Adds master

### PR DESCRIPTION
**Pull request by ezmiller**
@pletcher @AnnJohn opening this for an initial review. @pletcher you'll notice that there's no OAuth for github here. It felt like too much to me. Can you read through and see what you think. There's a lot of complexity here already. Happy to add it, if we end up thinking it necessary.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,344 @@
...
+
+Now, `encryptPassword`, as you can see, is a method that we've defined on the User model. That function returns a Promise that encrypts the user's supplied password using the bcrypt module. The promise either resolves with the hashed value of the password, or rejects, providing the error supplied by bcrypt as the reason.
+
+To check that this is working, add a new user by posting the appopriate data (i.e. name, username, email, password) to the `/user` route using either curl or Postman. Once you've created the user, check your database's users table. You should see a hash value in the password field that looks something like this:
+
+```
+$2a$10$pLOHxDVtdYQgemM2yVN.bOMTvWeMfRTV1ORgIlPPD0X9PBYHmPkCK 
+```
+
+Great, we're done configuring our schema.
```
Comment (pletcher) [inline README.md:136]: Could we include a paragraph here about why we don't ever want to store a plaintext password?

File: README.md [hunk 2]
```diff
@@ -1,2 +1,344 @@
...
+
+So what is our validation logic? Well, it's pretty straightforward it turns out. The callback we've supplied above takes the username and password, which the user will have supplied when they attempt to login, and then a callback that we've called done.  This callback is important because it is what we'll use to hand control to the next step in server's handling of the request. As we shall see, it has the following signature: `done(error, user[, msg])`.
+
+Inside the function we use our Bookshelf User model to try to fetch a user using the supplied username. Then, if the user is not found we call `done(null, false)` to indicate that there was no error (so `null` for the error argument), but no user was found (so `false` for the user argument). If a user was found, then we try to validate the password using the password validation function that we added to our User model. If the passowrd is invalid, we again call `done(null, false)`. Otherwise, we call `done(null, usr)`passing the usr that we've found to the next step in the process. Finally, if an error occurs we simply call `done(err)`.
```
Comment (pletcher) [inline README.md]: I like this -- could you add an explanation about why students shouldn't error out on an incorrect password but should instead (as you've done in this example) just return no user?

**Comment by pletcher** [comment]
This is a meaty lesson! I think it's nearly there -- just a couple of additions so that students understand the hows and wherefores of authentication a bit more intimately.

**Comment by ezmiller** [comment]
@pletcher I just made a few more minor changes here, a few spelling fixes. Where's this at otherwise?

**Comment by pletcher** [comment]
Looks good to me! I moved the one image to Flatiron's S3 bucket, but everything else seems 👍

**Pull request merged**